ReactiveCocoa/Swift/Signal.swift

```diff
@@ -17,8 +17,16 @@ import Result
 public final class Signal<T, E: ErrorType> {
 	public typealias Observer = SinkOf<Event<T, E>>
 
-	private let lock = NSRecursiveLock()
-	private var observers: Bag<Observer>? = Bag()
+	private let atomicObservers: Atomic<Bag<Observer>?> = Atomic(Bag())
+
+	/// Used to ensure that events are serialized during delivery to observers.
+	private let sendLock = NSLock()
+
+	/// Disposes of the generator provided at initialization time.
+	private let generatorDisposable = SerialDisposable()
+
+	/// When disposed of, the Signal should interrupt as soon as possible.
+	private let interruptionDisposable = SimpleDisposable()
```

Comment:
Ahh, it's hiding an `Atomic<Bool>`. Might be better to just do that directly since you're not putting this in a `CompositeDisposable`.